Fix detections of tagged packets.

The check for the 8021Q ethertype must ensure to use a pointer to uint16_t.
Add a typecast to the macro 'IS_TAGGED'
--- a/src/ieee8021ag.h
+++ b/src/ieee8021ag.h
@@ -209,7 +209,7 @@ struct cfm_lbm {
 #define GET_MD_LEVEL(s)		(((s)->octet1.md_level >> 5) & 0x07)
 
 /* positions of headers in Ethernet frame */
-#define IS_TAGGED(s)		(*(s + ETHER_ADDR_LEN * 2) \
+#define IS_TAGGED(s)		(*(uint16_t*)(s + ETHER_ADDR_LEN * 2) \
 					== htons(ETYPE_8021Q))
 #define CFMHDR(s)		(struct cfmhdr *) \
 				(IS_TAGGED(s) ? \
